<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>924</width>
    <height>795</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NCam Lens </string>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="NCamWidget" name="ncamWidget" native="true"/>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="LensWidget" name="lensWidget" native="true">
        <property name="minimumSize">
         <size>
          <width>512</width>
          <height>256</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Show Channels</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QCheckBox" name="zoomCheckBox">
           <property name="text">
            <string>Zoom (red)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="zoomLineEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="focusCheckBox">
           <property name="text">
            <string>Focus (green)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="focusLineEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="fovCheckBox">
           <property name="text">
            <string>Fov (blue)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="fovLineEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="LensImageWidget" name="lensImage" native="true">
        <property name="minimumSize">
         <size>
          <width>512</width>
          <height>256</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>924</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NCamWidget</class>
   <extends>QWidget</extends>
   <header>NCamWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LensWidget</class>
   <extends>QWidget</extends>
   <header>LensWidget.h</header>
   <container>1</container>
   <slots>
    <signal>tableUpdated()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>LensImageWidget</class>
   <extends>QWidget</extends>
   <header>LensImageWidget.h</header>
   <container>1</container>
   <slots>
    <slot>zoomChannel(int)</slot>
    <slot>focusChannel(int)</slot>
    <slot>fovChannel(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>aboutDialogShow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lensWidget</sender>
   <signal>tableUpdated()</signal>
   <receiver>MainWindow</receiver>
   <slot>onTableUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>689</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>lensImage</receiver>
   <slot>zoomChannel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>491</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>657</x>
     <y>636</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fovCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>lensImage</receiver>
   <slot>fovChannel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>823</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>657</x>
     <y>636</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>focusCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>lensImage</receiver>
   <slot>focusChannel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>657</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>657</x>
     <y>636</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>aboutDialogShow()</slot>
  <slot>fileNew()</slot>
  <slot>onTableUpdate()</slot>
  <slot>updateSample(double,double,double,double)</slot>
 </slots>
</ui>
